python/pdstools/app/datamart_utils.py

```diff
@@ -129,7 +128,8 @@ def get_from_s3(bucket, key, sql_query):
 def import_datamart(_fun, *args, **kwargs):
     dm = _fun(*args, **kwargs)
     dm.modelData = dm.modelData.collect()
-    dm.predictorData = dm.predictorData.collect()
+    if kwargs.get("predictor_df") is not None:
```

Comment:
Do we need to do this? I'd rather not

Reply:
I mean I'd rather not collect predictordata at all :D

Reply:
`@st.cache_resource(show_spinner=True)` caching was giving an error for LazyFrame but you already created a custom caching function for lazyframes I believe. Do we have it already in the repo?

Reply:
@st.cache(
    hash_funcs={pl.LazyFrame: lambda x: hash(x.explain(optimized=False))},
)